commit 85487cdd837f7c48ccb48cbb37e789eed7c0aa02
Author: Sergei Trofimovich <slyfox@gentoo.org>
Date:   Sat Apr 30 08:32:16 2016 +0100

    Hledger/UI/UITypes.hs: fix build failure agains latest regex-tdfa
    
    regex-tdfa used to import 'Text.Show.Functions' and stopped
    doing that. As a result a few packages exposed bugs trying
    to show functions (usually by accident).
    
    hledger-ui fails as:
    
      Hledger/UI/UITypes.hs:20:15:
        No instance for (Show Screen)
          arising from the third field of ‘AppState’ (type ‘Screen’)
        Possible fix:
          use a standalone 'deriving instance' declaration,
            so you can specify the instance context yourself
        When deriving the instance for (Show AppState)
    
    Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

diff --git a/Hledger/UI/UITypes.hs b/Hledger/UI/UITypes.hs
index a71a218..6cccbea 100644
--- a/Hledger/UI/UITypes.hs
+++ b/Hledger/UI/UITypes.hs
@@ -17,7 +17,7 @@ data AppState = AppState {
   ,ajournal :: Journal      -- ^ the journal being viewed
   ,aScreen :: Screen        -- ^ the currently active screen
   ,aPrevScreens :: [Screen] -- ^ previously visited screens, most recent first
-  } deriving (Show)
+  }
 
 -- | Types of screen available within the app, along with their state.
 -- Screen types are distinguished by their constructor and their state
@@ -52,6 +52,5 @@ data Screen =
     ,sHandleFn :: AppState -> V.Event -> EventM (Next AppState)
     ,sDrawFn :: AppState -> [Widget]
     }
-  deriving (Show)
 
 instance Show (List a) where show _ = "<List>"
